tf2_ros/docs/source/index.rst

```diff
@@ -1,44 +1,95 @@
-tf2_ros Overview
-================
+Overview
+========
 
-This is the Python API reference for the tf2_ros package.
+This package contains the ROS 2 bindings for the tf2 library, for both Python and C++.
 
-To broadcast transforms using ROS:
-- Call :meth:`rospy.init` to initialize a node.
-- Construct a :class:`tf2_ros.TransformBroadcaster`.
-- Pass a :class:`geometry_msgs.TransformStamped` message to :meth:`tf2_ros.TransformBroadcaster.sendTransform`.
+1.1 Broadcasting Transforms
+---------------------------
+  * **tf2_ros::TransformBroadcaster()**, constructor
+  * **tf2_ros::TransformBroadcaster::sendTransform** to send transforms
 
-    - Alternatively, pass a vector of :class:`geometry_msgs.TransformStamped` messages.
+Similarly static transforms can be sent by:
 
-To listen for transforms using ROS:
-- Construct an instance of a class that implements :class:`tf2_ros.BufferInterface`.
+  * **tf2_ros::StaticTransformBroadcaster()**, constructor,
+  * **tf2_ros::StaticTransformBroadcaster::sendTransform** to send static transforms
 
-    - :class:`tf2_ros.Buffer` is the standard implementation which offers a tf2_frames service that can respond to requests with a :class:`tf2_msgs.FrameGraph`.
-    - :class:`tf2_ros.BufferClient` uses an :class:`actionlib.SimpleActionClient` to wait for the requested transform to become available.
+1.2 Using Published Transforms
+------------------------------
+For most purposes using tf2_ros will be done using tf2_ros::Buffer. It's main public API is defined by tf2_ros::BufferInterface. Typically it will be populated using a tf2_ros::TransformListener which subscribes to the appropriate topics.
 
-- Pass the :class:`tf2_ros.Buffer` to the constructor of :class:`tf2_ros.TransformListener`.
-    - Optionally, pass a :class:`ros.NodeHandle` (otherwise TransformListener will connect to the node for the process).
-    - Optionally, specify if the TransformListener runs in its own thread or not.
+  * **tf2_ros::Buffer::transform** is the main method for applying transforms.
```

Comment:
Right now the loacl links are done through [rst](https://www.sphinx-doc.org/en/master/usage/referencing.html#ref-role) which puts them next to what we're trying to list. I assumed this would be the easiest thing to do when it came to overall documentation.
Here's a video of what it should look like:

https://github.com/ros2/geometry2/assets/82680922/2d91b8f1-45de-4631-92b3-1493758b2a5f

To recreate do the following in the terminal
1) Clone the repo
2) `cd tf2_ros/docs`
3) `make html`
4) `rosdoc2 build --package-path ./..`
5) `rosdoc2 open ./docs_output/tf2_ros/index.html`
6) Run it on the web so something like `python3 -m http.server` will work for localhost

Are packages like `rosdoc2`, `exhale`, and `breathe` able to run on the docs, or is this overengineered?